src/content/en/updates/2017/09/abortable-fetch.md

```diff
@@ -0,0 +1,306 @@
+
+    fetch(url, { signal }).then(response => {
+      return response.text();
+    }).then(text => {
+      console.log(text);
+    }).catch(err => {
+      if (err.name === 'AbortError') {
+        console.log('Fetch aborted');
+      } else {
+        console.log('Uh oh, an error!', err);
+      }
+    });
+
+You don't often want to show an error message if the user aborted the operation, as it isn't an
+"error" if you successfully do that the user asked. To avoid this, use an if-statement such as the
+one above to handle abort errors specifically.
+
+Here's an example that gives the user a button to load content, and a button to abort. If the fetch
+errors, an error is shown, *unless* it's an abort error:
+
+    // This will allow us to abort the fetch.
+    let controller;
+
+    // Abort if the user clicks:
+    abortBtn.addEventListener('click', () => {
+      if (controller) controller.abort();
+    });
+
+    // Load the content:
+    loadBtn.addEventListener('click', async () => {
+      controller = new AbortController();
+      const signal = controller.signal;
+
+      // Prevent another click until this fetch is done
+      loadBtn.disabled = true;
+      abortBtn.disabled = false;
+
+      try {
+        // Fetch the content & use the signal for aborting
+        const response = await fetch(contentUrl, { signal });
+        // Add the content to the page
+        output.innerHTML = await response.text();
+      }
+      catch (err) {
+        // Avoid showing an error message if the fetch was aborted
+        if (err.name !== 'AbortError') {
+          output.textContent = "Oh no! Fetching failed.";
+        }
+      }
+
+      // These actions happen no matter how the fetch ends
+      loadBtn.disabled = false;
+      abortBtn.disabled = true;
+    });
+
+Note: This example uses [async
+functions](/web/fundamentals/getting-started/primers/async-functions).
+
+**[Here's a demo](https://fetch-svg-abort.glitch.me/)** – At time of writing, the only browser which
+supports this is Firefox 57.
+
+## One signal, many fetches
+
+A single signal can be used to abort many fetches at once:
+
+    async function fetchStory({ signal }={}) {
+      const storyResponse = await fetch('/story.json', { signal });
+      const data = await response.json();
+
+      const chapterFetches = data.chapterUrls.map(async url => {
+        const response = await fetch(url, { signal });
+        return response.text();
+      });
+
+      return Promise.all(chapterFetches);
+    }
+
+In the above example, the same signal is used for the initial fetch, and for the parallel chapter
+fetches. Here's how you'd use `fetchStory`:
+
+    const controller = new AbortController();
+    const signal = controller.signal;
+
+    fetchStory({ signal }).then(chapters => {
+      console.log(chapters);
+    });
+
+In this case, calling `controller.abort()` will either abort whichever fetches are in-progress.
```

Comment:
This is a great example

Reply:
Cheers! "either abort whichever" tho, will fix that.